tests/unit/test_collector.py

```diff
@@ -0,0 +1,41 @@
+import unittest
+from unittest.mock import Mock, patch
+
+import pytest
+
+from prometheus_juju_backup_all_exporter.collector import CollectorBase
+
+
+class TestCollectorBase(unittest.TestCase):
+    """Collector base test class."""
+
+    def setUp(self):
+        self.patch_abstract_method = patch.object(
+            CollectorBase, "__abstractmethods__", set()
+        )
+        self.patch_abstract_method.start()
+        mock_metric = Mock()
+        mock_metric.__class__ = Mock()
+        mock_metric.add_samples = Mock()
+        self.test_subclass = CollectorBase(mock_metric)
+
+    def tearDown(self):
+        self.patch_abstract_method.stop()
```

Comment:
not-blocking:  FYI: You could use `self.addCleanup(self.patch_abstract_method.stop()` into `setUp`.

Reply:
DONE